src/NFTBill.sol

```diff
@@ -39,6 +40,29 @@ contract NFTBill is ERC1155 {
         _mint(msg.sender, id, 1, '');
     }
 
+    // To do a gasless withdraw we just need permission to transfer the NFT owner's
+    // ERC20 (and we could take some of it to pay the gas fees).
```

Comment:
Well this can allow gasless deposit, not withdraw! This allows the user to skip calling `approve` on ERC20 before calling deposit.

Reply:
Updated! Hopefully got it right/closer this time